exchange/bidder.go

```
@@ -522,11 +523,27 @@ func (bidder *bidderAdapter) doRequest(ctx context.Context, req *adapters.Reques
 
 func (bidder *bidderAdapter) doRequestImpl(ctx context.Context, req *adapters.RequestData, logger util.LogMsg, bidderRequestStartTime time.Time, tmaxAdjustments *TmaxAdjustmentsPreprocessed) *httpCallInfo {
 	var requestBody []byte
+	var gzipWriterPool = sync.Pool{
+		New: func() interface{} {
+			return gzip.NewWriter(nil)
+		},
+	}
 
 	switch strings.ToUpper(bidder.config.EndpointCompression) {
 	case Gzip:
-		requestBody = compressToGZIP(req.Body)
+		// Compress to GZIP
+		var b bytes.Buffer
+		w := gzipWriterPool.Get().(*gzip.Writer) // Utilize Sync Pool
+		w.Reset(&b)
+		w.Write([]byte(requestBody))
+		w.Close()
```

Please do not ignore errors from `Write` and `Close`.